PV260_Project/PresentationLayer/UI/DataSetUi.cs

```diff
@@ -18,84 +12,135 @@ public class DataSetUi : BaseUi, IDataSetUi
             _diffComputer = diffComputer;
             _dataLoader = dataLoader;
+            _dataDownloader = dataDownloader;
         }
 
-        public async Task Run() {
+        public async Task Run()
+        {
             await GenerateUi(
-                new List<MenuAction>() {
-                    new MenuAction() { Identifier = "1", Description = "Load file", Action = LoadFile},
-                    new MenuAction() { Identifier = "2", Description = "Download current file", Action = DownloadCurrentFile},
-                    new MenuAction() { Identifier = "3", Description = "View all files", Action = ViewFiles},
-                    new MenuAction() { Identifier = "4", Description = "Make a diff", Action = MakeDiff},
-                    new MenuAction() { Identifier = "5", Description = "Delete file by ID", Action = DeleteFile}
+                new List<MenuAction>
+                {
+                    new() { Identifier = "1", Description = "Load file", Action = LoadFile },
+                    new() { Identifier = "2", Description = "Download current file", Action = DownloadCurrentFile },
+                    new() { Identifier = "3", Description = "View all files", Action = ViewFiles },
+                    new() { Identifier = "4", Description = "Make a diff", Action = MakeDiff },
+                    new() { Identifier = "5", Description = "Delete file by ID", Action = DeleteFile }
                 });
         }
 
-        private async Task LoadFile() {
-            Console.WriteLine("Path: ('b' for back)");
-            var input = Console.ReadLine();
+        private async Task LoadFile()
+        {
+            var input = "";
 
-            while (input! != "b") {
-                if (!File.Exists(input)) {
+            while (input! != "b")
+            {
+                Console.WriteLine("Path: ('b' for back)");
+                input = Console.ReadLine();
+
+                if (!File.Exists(input))
+                {
                     Console.WriteLine("File on this path does not exist!");
                     input = Console.ReadLine();
                     continue;
                 }
 
-                var file = _dataLoader.LoadCsvFile(input);
-                await _dataSetService.CreateDataSet(file);
+                try
+                {
+                    var file = _dataLoader.LoadCsvFile(input);
+                    await _dataSetService.CreateDataSet(file);
+                }
+                catch (DataSetAlreadyExistsException e)
+                {
+                    Console.WriteLine("File from this date was already loaded!");
+                    continue;
+                }
+
                 return;
             }
         }
 
-        private async Task DownloadCurrentFile() {
+        private async Task DownloadCurrentFile()
+        {
+            var input = "";
 
+            while (input! != "b")
+            {
+                Console.WriteLine("Path: ('b' for back)");
```

Comment:
I see you got inspired by the `LoadFile` method :D but I'm not sure, if downloading file needs path. I think that the path should be written somewhere in code or in config so users won't need to remember the url. Also this method looks a lot like the `LoadFile`. Maybe refinement? :D If you are bored... :D

Reply:
I did not know how to implement it actually. I think this actually makes sense for manual downloading the files (cuz user can choose his own destination) and for automatic downloads, there would be a URL in DB.

Reply:
Ok, good point